dask/_task_spec.py

```diff
@@ -569,6 +574,10 @@ def __init__(self, key: Any, value: Any):
         self.typ = type(value)
         self._dependencies = _no_deps
 
+    @property
+    def data_producer(self) -> bool:
+        return True
```

Comment:
This is an interesting change that will affect scheduling behavior. Likely not in a meaningful way but it's not entirely obvious.

These kind of tasks were previously not considered as root-ish but now will be. The interesting part about these is that they will allocate memory not when they are executed but as soon as they are deserialized. Queuing therefore will postpone memory allocation until the task is actually needed which is arguable a saner behavior.

Since these tasks are typically few and are supposed to be still relatively small, I don't think this will be noticeable.

Reply:
Yeah, I agree. I played around with this a while ago when we where messing with the rootish thresholds, that didn't impact anything